nunaliit2-js/src/main/js/nunaliit2/n2.widgetLegend.js

```diff
@@ -28,558 +28,1306 @@ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+					node = document.createElementNS('http://www.w3.org/2000/svg', type);
+					if (id) {
+						node.setAttributeNS(null, 'id', id);
+					};
+				};
+				return node;    
+			},
+
+			_setAttr: function(node, name, value) {
+				node.setAttributeNS(null, name, value);
+			},
+
+			_addClass: function(elem, className) {
+				var classNames = [];
+
+				var currentClasses = elem.getAttribute('class') || '';
+				if( currentClasses ) {
+					classNames = currentClasses.split(' ');
+				};
+
+				if( classNames.indexOf(className) < 0 ){
+					classNames.push(className);
+				};
+
+				elem.setAttribute('class',classNames.join(' '));
+			},
+
+			/**
+			 * Method: _computePathFromSymbol Given an OpenLayers symbol (array of
+			 * points, which are tuples of x,y coordinates), create a SVG path with an
+			 * approximate area of 30 (area of a circle with a radius of 5) Example for
+			 * symbol: [0,0, 1,0, 1,1, 0,1, 0,0] Examplke fo SVG Path: 'M -4.4 -4.4 L
+			 * -4.4 4.4 L 4.4 4.4 L 4.4 -4.4 Z'
+			 */
+			_computePathFromSymbol: function(symbol){
+				var area = 0,
+				minx = undefined,
+				maxx = undefined,
+				miny = undefined,
+				maxy = undefined;
+
+				// Figure out bounding box
+				for(var i=0,e=symbol.length; i<e; i=i+2){
+					var x = symbol[i];
+					var y = symbol[i+1];
+
+					if( typeof minx === 'undefined' ){
+						minx = x;
+					} else if( minx > x ){
+						minx = x;
+					};
+
+					if( typeof maxx === 'undefined' ){
+						maxx = x;
+					} else if( maxx < x ){
+						maxx = x;
+					};
+
+					if( typeof miny === 'undefined' ){
+						miny = y;
+					} else if( miny > y ){
+						miny = y;
+					};
+
+					if( typeof maxy === 'undefined' ){
+						maxy = y;
+					} else if( maxy < y ){
+						maxy = y;
+					};
+				};
+
+				// Compute path, recentering the symbol and adjusting the area so
+				// it fits a bounding box of 10x10
+				var path = [],
+				transx = (minx+maxx)/2,
+				transy = (miny+maxy)/2,
+				width = maxx-minx,
+				height = maxy-miny,
+				factor = (width > height) ? width / 10 : height / 10;
+				if( factor <= 0 ){
+					factor = 1;
+				};
+				for(var i=0,e=symbol.length; i<e; i=i+2){
+					var x = symbol[i];
+					var y = symbol[i+1];
+
+					var effX = (x-transx)/factor;
+					var effY = (y-transy)/factor;
+
+					// Round to .01
+					effX = Math.floor(effX * 100) / 100;
+					effY = Math.floor(effY * 100) / 100;
+
+					if( 0 === i ){
+						path.push('M ');
+					} else {
+						path.push('L ');
+					};
+
+					path.push(''+effX);
+					path.push(' '+effY+' ');
+				};
+				path.push('Z');
+
+				return path.join('');
+			}
+			//=================================================================
+			
+			
+	})
+//	--------------------------------------------------------------------------
+	function HandleWidgetAvailableRequests(m){
+		if( m.widgetType === 'legendWidget' ){
+			if( $.fn.slider ) {
+				m.isAvailable = true;
+			};
+		} else if ( m.widgetType === 'legendWidget2' ){
```

Comment:
Can move this into first if as an OR (||) condition, since the m.isAvailable = true in both cases.

Reply:
In nunaliit, conventionally, each js-module only has one event handler function. So all the events that this very js-module listening on, jump into this function. Therefore, for maintenance cause, it is better to limit all the actions inside a if...else if scope. Keep in mind, just for clarity, not for code isolation, because js doesn't have braces local scope.